Geometry_Engine/Compute/GrahamScan.cs

```diff
@@ -0,0 +1,151 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Quantities.Attributes;
+using BH.Engine.Base;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Geometry
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Implements the GrahamScan algorithm to determine the convex hull of a list of points contained within the XY Plane.")]
+        [Input("pts", "The points to determine the convex hull contained within the XY Plane.")]
+        [Input("tolerance", "Geometrical tolerance to be used in the method.", typeof(Length))]
+        [Output("c", "The convex hull of the point list, no repeat points are returned.")]
+        public static List<Point> GrahamScan(List<Point> pts, double tolerance = Tolerance.MicroDistance)
```

Comment:
I had thought `MicroDistance` was 1E-03, my mistake! Have updated to `Tolerance.MacroDistance` - I generally use closest mm.

Reply:
I would be careful here tbh - for structural elements I think we always use `Tolerance.Distance` because 1e-3 may be not sensitive enough (would be good to have @IsakNaslundBh around to confirm), so using 1e-3 here would be a bit of a risk factor + an oddity too IMO. But the decision is yours, feel free to go either way as long as it works for you 👍

Reply:
It's ultimately a method in the Geometry_Engine, so happy to adopt whatever default is used on other methods - looks like be `Tolerance.Distance`.